rustls/src/crypto/hpke.rs

```diff
@@ -0,0 +1,85 @@
+use alloc::boxed::Box;
+use alloc::vec::Vec;
+use core::fmt::Debug;
+
+use crate::msgs::enums::HpkeKem;
+use crate::msgs::handshake::HpkeSymmetricCipherSuite;
+use crate::Error;
+
+/// A provider for [RFC 9180] Hybrid Public Key Encryption (HPKE) in base mode.
+///
+/// At a minimum each provider must support the [HPKE ciphersuite profile] required for
+/// encrypted client hello (ECH):
+///  * KEM: DHKEM(X25519, HKDF-SHA256)
+///  * symmetric ciphersuite:  AES-128-GCM w/ HKDF-SHA256
+///
+/// [RFC 9180]: <https://www.rfc-editor.org/rfc/rfc9180.html>
+/// [HPKE ciphersuite profile]: <https://datatracker.ietf.org/doc/html/draft-ietf-tls-esni-17#section-9>
+pub trait HpkeProvider: Debug + Send + Sync + 'static {
+    /// Create a base mode HPKE instance for the chosen suite.
+    ///
+    /// May return an error if the suite is unsupported by the provider.
+    fn create(&self, suite: &HpkeSuite) -> Result<Box<dyn Hpke>, Error>;
```

Comment:
At risk of bikeshedding even more, maybe `start()`? `create()` still feels to constructorish to me.

Reply:
Switched to `start` and adapted the rustdoc comment.